src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -64,6 +66,39 @@ internal struct NameDeclarationInfo
                 return default;
             }
 
+            private static bool IsTupleElementDeclaration(
+                SyntaxToken token, SemanticModel semanticModel, int position,
+                CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                result = IsFollowingTypeOrComma<TupleElementSyntax>(
+                    token,
+                    semanticModel,
+                    tupleElement => tupleElement.Type,
+                    _ => default(SyntaxTokenList),
+                    _ => ImmutableArray.Create(SymbolKind.Local), cancellationToken);
+
+                return result.Type != null;
+            }
+
+            private static bool IsTupleExpressionDeclaration(
+                SyntaxToken token, SemanticModel semanticModel, int position,
+                CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                if (token.GetAncestor(node => node.IsKind(SyntaxKind.TupleExpression)) != null)
```

Comment:
This ancestor check feels too loose. Imagine an invocation inside a tuple expression, and the caret is after the comma in the argument list of the invocation.
Can we just check the parent or grandparent instead?

Reply:
I tried to come up with a test to make this fail but I couldn't. Can you have a look in commit 3ebfe28e? We somehow need to get a type declaration inside that call to make the completion fail. The method examines is for code like this and I don't see how there can be a method invocation:

```CS
class Test
{
    void Do()
    {
        (System.Array array, System.Action $$ 
    }
}
```

Reply:
See also my previous discussion with @sharwell https://github.com/dotnet/roslyn/pull/22673#discussion_r175764190